<?xml version="1.0" ?>
<launch>
  <arg name="namespace"		default="dynamixel_controller_1"/>
  <arg name="usb_port"		default="/dev/ttyUSB0"/>
  <arg name="dxl_baud_rate"	default="1000000"/>

  <arg name="use_moveit"	default="false"/>
  <arg name="use_joint_state"	default="true"/>
  <arg name="use_cmd_vel"	default="false"/>

  <arg name="screw_tools"	default="$(dirname
					 )/../config/$(arg screw_tools).yaml"/>

  <param name="dynamixel_info"	value="$(arg screw_tools)"/>

  <node name="$(arg namespace)"
	pkg="dynamixel_workbench_controllers"
	type="dynamixel_workbench_controllers"
        output="screen" args="$(arg usb_port) $(arg dxl_baud_rate)">
    <param name="use_moveit"		 value="$(arg use_moveit)"/>
    <param name="use_joint_states_topic" value="$(arg use_joint_state)"/>
    <param name="use_cmd_vel_topic"	 value="$(arg use_cmd_vel)"/>
    <rosparam>
      publish_period: 0.010
      dxl_read_period: 0.010
      dxl_write_period: 0.010
      mobile_robot_config:                <!--this values will be set when 'use_cmd_vel' is true-->
      seperation_between_wheels: 0.160  <!--default value is set by reference of TB3-->
      radius_of_wheel: 0.033            <!--default value is set by reference of TB3-->
    </rosparam>
  </node>

  <node name="screw_tool_controller"
	pkg="aist_fastening_tools" type="screw_tool_controller.py"
	output="screen">
    <param name="controller_ns"		value="$(arg namespace)"/>
    <rosparam param="screw_tools"	file="$(arg screw_tools)"/>
  </node>

</launch>
